### DISCUSSION TITLE
what is the equivalent of `python setup.py --version` now that setup.py CLI is deprecated?

### DISCUSSION THREAD

**Discussion by lsh-0** [pypa/setuptools#3958] [General]
I'm sorry if this has been asked, but I can't find an answer in the issues or discussions.

What is the equivalent of `python setup.py --version` now that the setup.py CLI is deprecated?

My setup.py file has a `Version='4.0.0'`: https://github.com/elifesciences/threadbare/blob/cd5669acf5889ea831983693f248fe6d2b37aa44/setup.py#L5

and `python setup.py --version` prints this value out as-is:

```bash
$ python setup.py --version
4.0.0
```

Very simple, but I can't find an alternative and don't want to resort to grepping and regex. Any help?

**Comment by webknjaz** [comment]
Depending on what you need this for, the answer would differ. You could install the thing and get it from meta through `importlib.metadata`. Technically, you could try asking PEP 517 build front-ends to implement this or call the corresponding hooks yourself...

**Comment by lsh-0** [comment]
Parsing the semver version from the wheel does the trick:

    ls dist/*.whl | grep -o -E '([0-9]+\.[0-9]+\.[0-9]+)'
